frontend/src/app/modules/members/members-page/members-page.component.ts

```diff
@@ -0,0 +1,144 @@
+import { RoleService } from '@core/services/role.service';
+import { ShareDataService } from '@core/services/share-data.service';
+import { ToastNotificationService } from '@core/services/toast-notification.service';
+import { Member } from '@shared/models/member/member';
+import { MemberItem } from '@shared/models/member/member-item';
+import { Role } from '@shared/models/role/role';
+import { TeamOption } from '@shared/models/teams/team-option';
+import { User } from '@shared/models/user/user';
+import { TreeNode } from 'primeng/api';
+
+@Component({
+    selector: 'app-members-page',
+    templateUrl: './members-page.component.html',
+    styleUrls: ['./members-page.component.sass']
+})
+
+export class MembersPageComponent extends BaseComponent implements OnInit {
+    loadingNumber = 0;
+    memberItems: MemberItem[];
+    isInviting: Boolean;
+    user: User;
+    roles: Role[];
+
+    isEdit: boolean = false;
+
+    constructor(
+        private memberService: MemberService,
+        private roleService: RoleService,
+        private toastNotifications: ToastNotificationService,
+        private authService: AuthenticationService,
+        private confirmWindowService: ConfirmWindowService,
+        private updateDataService: ShareDataService<Member>
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.user = this.authService.getUser();
+        this.isInviting = false;
+        this.loadingNumber += 1;
+        this.authService.getOrganization()
+            .pipe(this.untilThis)
+            .subscribe(organization => {
+                this.memberService.getMembersByOrganizationId(organization.id)
+                    .pipe(this.untilThis)
+                    .subscribe(members => {
+                        this.memberItems = members.map(member => ({ member, treeTeams: this.fromTeams(member.teams) }));
+                        this.loadingNumber -= 1;
+                        this.updateDataService.currentMessage
+                            .pipe(this.untilThis)
+                            .subscribe(member => {
+                                this.memberItems = this.memberItems.concat({ member, treeTeams: this.fromTeams(member.teams) });
+                            });
+                    }, error => {
+                        this.toastNotifications.error(error.toString());
+                        this.loadingNumber -= 1;
+                    });
+            });
+
+        this.roleService.getRoles()
+            .pipe(this.untilThis)
+            .subscribe(roles => {
+                this.roles = roles;
+            });
+    }
+
+    fromTeams(teams: TeamOption[]): TreeNode[] {
+        return [{
+            label: 'Teams',
+            children: teams.map(team => ({
+                label: team.name
+            }))
+        }];
+    }
+
+    startEdit(memberItemInput: MemberItem) {
+        const memberItem = memberItemInput;
+        memberItem.isEdit = true;
```

Comment:
Please get rid of mutation

Reply:
It seems that she uses this field to manipulate flow of UI, that's not the data, that will be sended or obtained by backend or frontend services, should she really rid of mutation in it? (that's same thing as isLoading = false as I can see)